### PR TITLE
Adding a plural namer to go2idl

### PR DESCRIPTION
**Pull request by caesarxuchao**
ref #17468

### CODE REVIEW

File: cmd/libs/go2idl/client-gen/generators/client-generator.go [hunk 1]
```diff
@@ -32,10 +32,16 @@ import (
 
 // NameSystems returns the name system used by the generators in this package.
 func NameSystems() namer.NameSystems {
+	pluralExceptions := map[string]string{
+		"Endpoints":       "endpoints",
+		"ComponentStatus": "componentStatus",
```
Comment (wojtek-t) [inline cmd/libs/go2idl/client-gen/generators/client-generator.go]: those looks a bit strange - can we have both capitalized or both not-capitalized?

File: cmd/libs/go2idl/namer/plural_namer.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import "k8s.io/kubernetes/cmd/libs/go2idl/types"
+
+type pluralNamer struct {
+	exceptions map[string]string
+	finalize   func(string) string
+}
+
+// NewPluralNamer returns a namer that returns the plural form of the input
+// type's name.
+func NewPluralNamer(exceptions map[string]string, public bool) *pluralNamer {
```
Comment (wojtek-t) [inline cmd/libs/go2idl/namer/plural_namer.go]: To be honest, I vote for creating two separate New functions:

func NewPublicPluralNamer(exceptions)
func NewPrivatePluralNamer(exceptions)

having bool in such methods is always a bit more confusing.

WDYT?

**Comment by wojtek-t** [comment]
Generally looks ok - just two minor comments.

File: cmd/libs/go2idl/client-gen/generators/client-generator.go (hunk 1 above)
Comment (caesarxuchao) [inline cmd/libs/go2idl/client-gen/generators/client-generator.go]: I capitalized both.

File: cmd/libs/go2idl/namer/plural_namer.go (hunk 2 above)
Comment (caesarxuchao) [inline cmd/libs/go2idl/namer/plural_namer.go]: I agree. I changed it to what you suggested.

File: cmd/libs/go2idl/namer/plural_namer.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	finalize   func(string) string
+}
+
+// NewPublicPluralNamer returns a namer that returns the plural form of the input
+// type's name, starting with a uppercase letter.
+func NewPublicPluralNamer(exceptions map[string]string) *pluralNamer {
+	return &pluralNamer{exceptions, IC}
+}
+
+// NewPrivatePluralNamer returns a namer that returns the plural form of the input
```
Comment (wojtek-t) [inline cmd/libs/go2idl/namer/plural_namer.go]: s/NewPluralNamer/NewPrivatePluralNamer/

**Comment by caesarxuchao** [comment]
Comments addressed. PTAL. Thank you.

**Comment by wojtek-t** [comment]
Just a very minor comment - feel free to self-apply lgtm label after applying.

File: cmd/libs/go2idl/namer/plural_namer.go (hunk 3 above)
Comment (caesarxuchao) [inline cmd/libs/go2idl/namer/plural_namer.go]: Sorry, I didn't think you will take a look that quickly:)   I fixed it in the newest commit.

**Comment by caesarxuchao** [comment]
Thank you. I'm going to apply the label according to a previous comment.

**Comment by wojtek-t** [comment]
This doesn't impact any production code - adding e2e-not required label.

**Pull request merged**